### ISSUE TITLE
Cloudflare caching

### ISSUE THREAD

**Issue by ethanresnick** [TechAtNYU/tech-nyu-site#58]
prompted by @thebyrd's [comment](https://github.com/TechAtNYU/tech-nyu-site/issues/57#issuecomment-76794092) about cloudflare caching:

1. can we ping cloudflare when we redeploy to invalidate its cache?
2. just more generally, can we control the cloudflare ttl? we need to think about this for the updating event on our homepage

**Comment by thebyrd** [comment]
1. Yes
2. Yes, I've made it 4 days, but can change that for you. If you're only updating html for each event, probs ok to make it 1 day?

**Comment by ethanresnick** [comment]
Cool.

Re 2. Sometimes we get our events into the system pretty last minute, like ~6 hours before (not ideal, but hey). Also, during Startup Week, where we can have one event right after another, we kinda need the page to update in near-real time. So, if we had to pick one time that would cover the SW case too, we'd probably want something really aggressive, like every 15 minutes. Is that an option on whatever free tier we're using? If not, that goes to the next point...

Re 1. How do we set that up?

**Comment by thebyrd** [comment]
1. There's an API I'll cook up a bash script for you

2. You guys are on the bowery business plan, so you can do all of this. I'd rather keep the TTL long and then just purge every time you deploy. Since people don't use the website that frequently, you loose a lot of the benefits of the cache if everything expires every 6 hours.

**Comment by thebyrd** [comment]
This repo is public and it involved an API secret, so I'll send it to you and Abhi in slack. Don't show it to anyone. It's a simple curl statement.

**Comment by ethanresnick** [comment]
@thebyrd Thanks David!

Keeping the TTL long and then purging only when necessary makes sense. I think that means purging in two cases:

1. On deploy, which is something that Abhi can easily add to the githook. I made [an issue](https://github.com/TechAtNYU/tech-nyu-site/issues/59) for that.

1. Every time the contents of `http://api.tnyu.org/v1.0/up-next-publicly` change. Since the site is [already polling](https://github.com/TechAtNYU/tech-nyu-site/blob/master/src/server.ls#L59) that url to get the next event, it can [pretty easily](https://github.com/TechAtNYU/tech-nyu-site/commit/c9d20ddfc71e1e21d8b7980c64e04952698f67fb) keep track of whether the next upcoming event has changed, and then alert cloudflare. But, when it does that, it **only needs to clear cloudflare's cache of the html file**, not the css or js. Is there an easy way to do that?

**Comment by ethanresnick** [comment]
Fuck. Moving to Cloudflare broke all our subdomains [again](https://github.com/TechAtNYU/tech-nyu-site/issues/38#issuecomment-56610516), so I had to temporarily switch the nameservers back to namecheap.

@thebyrd can you make sure cloudflare is reflecting all our other dns records properly (see below/in the namecheap control panel) before switching back

![screen shot 2015-03-03 at 6 09 23 pm](https://cloud.githubusercontent.com/assets/471894/6474764/ade9c434-c1d1-11e4-8ffb-f9dba98a7818.png)

**Comment by thebyrd** [comment]
:fire: :fire: 

adding a subdomain in cloudflare is pretty straightforward. A slack message would have solved the problem in 5 minutes. Now the site is down, when you're ready to get the site back up, please slack me the subdomains and ips you'd like.

**Comment by abhiagar** [comment]
Don't keep switching dns please! Breaks a lot of things during events

**Comment by thebyrd** [comment]
I haven't touched it yet

**Comment by ethanresnick** [comment]
@thebyrd You're right that I could have asked you first, and then only made the change on my own if you didn't respond right away. I just felt a sense of urgency because I got a message from the bd team telling me that they were sending out sponsorship prospectuses and companies couldn't get them to load, so I reacted in a bit of a panic. I do appreciate your help with this. 

@AbhiAgarwal Sorry that this broke stuff during the event! I just wanted to get the things back online for bd asap.

@
